### PR TITLE
relates project notes to current phase and displays badge

### PR DESCRIPTION
**Pull request by tillyw**
## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/10679
fixes https://github.com/cityofaustin/atd-data-tech/issues/11082

this pr stores the current phase of a project in the `moped_proj_notes` and table and displays it next to the associated status update or project comment. next pr will allow users to edit the associated phase!

## Testing
**URL to test:** 

[10679_relate_project_notes_to_current_phase--atd-moped-main.netlify.app/moped/](https://10679_relate_project_notes_to_current_phase--atd-moped-main.netlify.app/moped/)

**Steps to test:**
- create a new project
- add a status update named 'potential' (for easier tracking during testing)
- the new status update should appear with the 'potential' badge next to it (the modal will close automatically)
- add a comment with the same name
- the new comment should appear with the 'potential' badge next to it
- change the current phase under the 'timeline' tab
- add a new status update with the name of that phase
- check to make sure the badge reflects the update
- repeat the last pairs of steps with a comment
- delete a comment and/or status update, the appropriate ones should be deleted and the rest should persist

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryStatusUpdate.js
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryStatusUpdate.js:39]: since the only property you care about in `data` is the current phase ID, it would be better to require that prop explicitly, eg `currentPhaseId` instead of the generic `data` blob.

also, the `DasboardView` also uses this component and needs to be updated in the same way.

**Review by johnclary — CHANGES_REQUESTED** [review]
it's looking good! let me know if you have any questions about this feedback 👇

**Comment by tillyw** [comment]
thanks, @johnclary! will address these comments now, removing review requests from other others in the meantime.

**Comment by tillyw** [comment]
@johnclary updated to reflect your feedback and fix a bug! rerequesting review from everyone here. thanks!

**Comment by frankhereford** [comment]
This is squarely outside of the scope of this, but making a note of something I've observed while testing:

Screenshot 2023-01-04 at 5 22 02 PM

Screenshot 2023-01-04 at 5 22 14 PM

@johnclary An issue for consideration - should we unify the color of the font in the Quill WYSIWYG editor between when one is creating a new comment vs. when one is editing one?

File: moped-editor/src/views/projects/projectView/ProjectComments.js
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectComments.js:99]: ahh—i see now that `props.data` is being passed through the `ProjectView` component and can't easily be avoided. thanks for adding that `currentPhaseId` for `ProjectSummaryStatusUpdate` and adding the comment!

**Review by johnclary — APPROVED** [review]
@tillyw thanks for making such quick work of this. i'm really excited about this feature! 🚢 🚢 🚢

**Comment by johnclary** [comment]
> should we unify the color of the font in the Quill WYSIWYG editor between when a user is creating a new comment vs. when a user is editing a comment?

@frankhereford thanks for flagging that — will file a backlog issue 🙏

File: moped-editor/src/views/projects/projectView/ProjectComments.js
Comment (tillyw) [reply moped-editor/src/views/projects/projectView/ProjectComments.js:99]: yeah, i went back and forth on this multiple times - this was how i originally did it before i submitted my initial pr. thanks for helping me pick an approach!!

File: moped-database/migrations/1672851579949_set_fk_public_moped_proj_notes_phase_id/up.sql
Comment (frankhereford) [inline moped-database/migrations/1672851579949_set_fk_public_moped_proj_notes_phase_id/up.sql:6]: I'm wondering about the `on update cascade` and `on delete cascade` usage here, and I am not sure that want that behavior. 

I believe, as written, this says:

> When the `phase_id` of a `moped_phases` record is changed, go into `moped_proj_notes` and find all the records that point to the old ID and update them to point to the new one. 

> When a record in `moped_phases` is deleted, go into `moped_proj_notes` and delete any records that point to the being-deleted `moped_phases` record.

✅ The first behavior, the `on update` one, seems pretty great to me. I can't imagine a world when we'd be changing `phase_id`s, but it's very technically correct. 

🤔 The second behavior gives me more pause. I think we'd want the query to fail if we tried to remove a referred-to phase so we could be very intentional in how we want to handle those soon-to-be orphaned `moped_proj_notes` records. I'm pretty sure we wouldn't want to just delete them.

See next comment, for some more details, outside of this general observation.

File: moped-editor/src/queries/comments.js
Comment (mddilley) [inline moped-editor/src/queries/comments.js:20]: 🥳

File: moped-database/migrations/1672851579949_set_fk_public_moped_proj_notes_phase_id/up.sql
Comment (frankhereford) [reply moped-database/migrations/1672851579949_set_fk_public_moped_proj_notes_phase_id/up.sql:6]: In practice, right now, today, and probably not forever, we're safe-ish, because we have /another/ constraint named `moped_proj_phases_moped_phases_phase_id_fk` on the `moped_proj_phases` table which maintains the relationship between the value in the `phase_id` column in `moped_proj_phases` and the FK `moped_phases(phase_id)`. 

This constraint does not have `on update` or `on delete` behavior defined, so it's this constraint that prohibits the deletion of a referred to `moped_phase` record which would cascade on to the `moped_proj_notes` table as directed by the constraint in this PR.

Comment (frankhereford) [reply moped-database/migrations/1672851579949_set_fk_public_moped_proj_notes_phase_id/up.sql:6]: My recommendation is to remove the `on delete cascade` directive so we don't cascade into deleting users comments because we remove a phase without us first hand-crafting a resolution.

I think a low priority spin-off issue would be helpful to review these constraints and make sure we've set in directives that align to what we want to delete or not and when and also to ensure that we don't have conflicting or interacting constraints that prevent each other from doing what we told them to do. The constraint in this PR enters us into that state now but ...

The whole thing is largely moot because we almost universally soft-delete records and in this case, the records that have updates and deletes cascade from them are records that are "part of the app" and not user entered data, and as a result, are very stable. 

A final thought - this is very hard for me to get all loaded into my mind and to keep straight. It's complicated! Please forgive me for what I get wrong, but FWIW, these observations are borne out of testing in my local DB.

File: moped-editor/src/views/projects/projectView/ProjectComments.js
Comment (mddilley) [reply moped-editor/src/views/projects/projectView/ProjectComments.js:99]: +1 the comments and variable names make this super readable 🙏

**Review by mddilley — APPROVED** [review]
Test steps worked great - this is such a cool feature that really levels up comments with the context of what phase it was at the time! 🚀 🚀 🚀

**Review by frankhereford — APPROVED** [review]
Edit: 1st, I have to agree that this is a killer feature. It's so excellent in how well it conveys the state of the project when a comment was made, and it's just wonderfully clear and digestible. The design and implementation are top shelf. 

I tested this, including making comments when there is not an active phase, and it did just what I wanted it to. 💯

I tried really hard to write a concise explanation of my thoughts on the `on update` and `on delete` directives in the FK relationship. 😣 Sorry y'all -  It ended up kinda long, but we have a bit of a tangle here. 

It's a bit of a tangle that we should work out, but I don't think it needs to block this from shipping, except for I do think we should consider the `on delete cascade` again just in case we forget about circling back on it. My reasoning is below, and even as is, by happenstance, this PR does not put us in a danger zone, only danger-zone adjacent.

File: moped-database/migrations/1672851579949_set_fk_public_moped_proj_notes_phase_id/up.sql
Comment (tillyw) [reply moped-database/migrations/1672851579949_set_fk_public_moped_proj_notes_phase_id/up.sql:6]: thanks, @frankhereford ! i fully agree and will push an update that remove `on delete cascade` from the migration. for context, i was copying the behavior of the other fkey relationship in this table, `project_id`. that may be the desired behavior in that case, but i wanted to point that out in case we want to reevaluate that.

Screen Shot 2023-01-04 at 7 51 26 PM

**Comment by tillyw** [comment]
thanks y'all for the quick review! i'll leave it up to @johnclary's discretion as to whether to merge this tomorrow :)

**Comment by johnclary** [comment]
thanks @frankhereford for catching that and @tillyw for patching it right quick! totally agree we do not want to cascade delete from lookup tables. i've been opting for `set null` on the delete rules for optional foreign keys like this one, so i went ahead and patched that in.

i agree it'd be good to go through all our fk constraints and make sure we're being consistent. we've also got [#10525](https://github.com/cityofaustin/atd-data-tech/issues/10525) which would add another layer of protection by making sure users cannot edit lookup tables (like `moped_phases`) through the API.

**Pull request merged**
